backend/ml_utils.py

```
@@ -0,0 +1,338 @@
+import numpy as np
+import pandas as pd
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+import pickle
+import json
```

Ruff (0.12.2)

`json` imported but unused

Remove unused import: `json`

(F401)